src/uploader.rs

```diff
@@ -195,6 +219,20 @@ async fn upload_layer(
                 }
             };
 
+            open_file = match local_layer_path.clone() {
```

Comment:
great!

Reply:
I would say also try running `cargo clippy` (look it up you might need to install it) to see whether it will complain about more idiomatic way to do this